### DISCUSSION TITLE
brew cask install onyx checksum_does_not_match_error | Mac catalina

### DISCUSSION THREAD

**Discussion by whoooami** [Homebrew/discussions#2459] [Casks]
brew cask install onyx                                
==> Downloading https://www.titanium-software.fr/download/1015/OnyX.dmg
######################################################################## 100.0%
==> Verifying SHA-256 checksum for Cask 'onyx'.
==> Note: Running `brew update` may fix SHA-256 checksum errors.
Error: Checksum for Cask 'onyx' does not match.
Expected: 8c0ab6fbc0ade0444f34aad7d21c3c9274acf0b597a20c8f2311547a52bc74e9
  Actual: 0dd8119ad3441c5130ca584ac90ad450c272aab8b577925561a2536da48d2d54
    File: /Users/~/Library/Caches/Homebrew/downloads/b54a31b30b31f5f0c51f0aef445dcec70d3e82aaf7bdb33b94d485cb6ba7fc6f--OnyX.dmg
To retry an incomplete download, remove the file above.
If the issue persists, visit:
  https://github.com/Homebrew/homebrew-cask/blob/HEAD/doc/reporting_bugs/checksum_does_not_match_error.md

**Comment by SMillerDev** [comment]
Did you follow:
> If the issue persists, visit:
https://github.com/Homebrew/homebrew-cask/blob/HEAD/doc/reporting_bugs/checksum_does_not_match_error.md

**Reply by whoooami** [reply]
> Did you follow:
> 
> > If the issue persists, visit:
> > https://github.com/Homebrew/homebrew-cask/blob/HEAD/doc/reporting_bugs/checksum_does_not_match_error.md

Yeah, the checksum is different.

**Reply by neutric** [reply]
```
%  curl https://www.titanium-software.fr/download/1015/OnyX.dmg | sha256sum
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 4598k  100 4598k    0     0  4821k      0 --:--:-- --:--:-- --:--:-- 4871k
0dd8119ad3441c5130ca584ac90ad450c272aab8b577925561a2536da48d2d54  -
```
and `0dd8119ad3441c5130ca584ac90ad450c272aab8b577925561a2536da48d2d54` is exactly what is expected for Catalina as per current cask definition.

**Reply by whoooami** [reply]
> 
Year, but in brew it was 8c0ab6fbc0ade0444f34aad7d21c3c9274acf0b597a20c8f2311547a52bc74e9.

**Reply by carlocab** [reply]
Not sure where you're getting that checksum, but the cask has `0dd8119ad3441c5130ca584ac90ad450c272aab8b577925561a2536da48d2d54`.

You can see for yourself at https://github.com/Homebrew/homebrew-cask/blob/fbb42995b419687d46449a7b5b329d79d4060ab4/Casks/onyx.rb#L17-L19.

**Comment by gromgit** [comment]
@whoooami, the checksum you posted was for onyx v3.8.5, which was updated to v3.8.6 on **Sep 28 2020**, i.e. over a year ago. You might want to `brew update` before trying the install again. If that still doesn't work, post the output of `brew config` and `brew doctor`.

**Comment by whoooami** [comment]
@gromgit  Ok, I just run brew cask install onyx without version.

-- brew config
`
                                 HOMEBREW_VERSION: 2.5.2
                                ORIGIN: https://mirrors.aliyun.com/homebrew/brew.git
                                HEAD: cd4a3c0588abb82d916bc8558f4c157e951d449d
                                Last commit: 1 year, 2 months ago
                                Core tap ORIGIN: https://mirrors.aliyun.com/homebrew/homebrew-core.git
                                Core tap HEAD: 136a43943a8bb8dc47cba8f25bf0e73e29b3ffe1
                                Core tap last commit: 1 year, 2 months ago
                                Core tap branch: master
                                HOMEBREW_PREFIX: /usr/local
                                HOMEBREW_CASK_OPTS: []
                                HOMEBREW_MAKE_JOBS: 4
                                HOMEBREW_NO_AUTO_UPDATE: set
                                Homebrew Ruby: 2.6.3 => /usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3_2/bin/    ruby
                                CPU: quad-core 64-bit broadwell
                                Clang: 12.0 build 1200
                                Git: 2.24.3 => /Applications/Xcode.app/Contents/Developer/usr/bin/git
                                Curl: 7.64.1 => /usr/bin/curl
                                Java: 1.8.0_51
                                macOS: 10.15.7-x86_64
                                CLT: N/A
                                Xcode: 12.4
`

-- brew doctor
`
        Please note that these warnings are just used to help the Homebrew maintainers
        with debugging if you file an issue. If everything you use Homebrew for is
        working fine: please don't worry or file an issue; just ignore this. Thanks!
        
        Warning: Suspicious https://github.com/Homebrew/brew git origin remote found.
        The current git origin is:
          https://mirrors.aliyun.com/homebrew/brew.git
        
        With a non-standard origin, Homebrew won't update properly.
        You can solve this by setting the origin remote:
          git -C "/usr/local/Homebrew" remote set-url origin https://github.com/Homebrew/brew
        
        Warning: Suspicious https://github.com/Homebrew/homebrew-core git origin remote found.
        The current git origin is:
          https://mirrors.aliyun.com/homebrew/homebrew-core.git
        
        With a non-standard origin, Homebrew won't update properly.
        You can solve this by setting the origin remote:  

          git -C "/usr/local/Homebrew/Library/Taps/homebrew/homebrew-core" remote set-url origin        https://github.com/Homebrew/homebrew-core
        
        Warning: You have the following deprecated, cask taps tapped:
          caskroom/cask
        Untap them with `brew untap`.
        
        Warning: Unbrewed dylibs were found in /usr/local/lib.
        If you didn't put them there on purpose they could cause problems when
        building Homebrew formulae, and may need to be deleted.
        
        Unexpected dylibs:
          /usr/local/lib/libprotobuf-lite.10.dylib
          /usr/local/lib/libprotobuf.10.dylib
          /usr/local/lib/libprotoc.10.dylib
        
        Warning: Unbrewed header files were found in /usr/local/include.
        If you didn't put them there on purpose they could cause problems when
        building Homebrew formulae, and may need to be deleted.
        
        Unexpected header files:
          /usr/local/include/google/protobuf/any.h
          /usr/local/include/google/protobuf/any.pb.h
          /usr/local/include/google/protobuf/api.pb.h
          /usr/local/include/google/protobuf/arena.h
          /usr/local/include/google/protobuf/arenastring.h
          /usr/local/include/google/protobuf/compiler/code_generator.h
          /usr/local/include/google/protobuf/compiler/command_line_interface.h
          /usr/local/include/google/protobuf/compiler/cpp/cpp_generator.h
          /usr/local/include/google/protobuf/compiler/csharp/csharp_generator.h
          /usr/local/include/google/protobuf/compiler/csharp/csharp_names.h
          /usr/local/include/google/protobuf/compiler/importer.h
          /usr/local/include/google/protobuf/compiler/java/java_generator.h
          /usr/local/include/google/protobuf/compiler/java/java_names.h
          /usr/local/include/google/protobuf/compiler/javanano/javanano_generator.h
          /usr/local/include/google/protobuf/compiler/js/js_generator.h
          /usr/local/include/google/protobuf/compiler/objectivec/objectivec_generator.h
          /usr/local/include/google/protobuf/compiler/objectivec/objectivec_helpers.h
          /usr/local/include/google/protobuf/compiler/parser.h
          /usr/local/include/google/protobuf/compiler/plugin.h
          /usr/local/include/google/protobuf/compiler/plugin.pb.h
          /usr/local/include/google/protobuf/compiler/python/python_generator.h
          /usr/local/include/google/protobuf/compiler/ruby/ruby_generator.h
          /usr/local/include/google/protobuf/descriptor.h
          /usr/local/include/google/protobuf/descriptor.pb.h
          /usr/local/include/google/protobuf/descriptor_database.h
          /usr/local/include/google/protobuf/duration.pb.h
          /usr/local/include/google/protobuf/dynamic_message.h
          /usr/local/include/google/protobuf/empty.pb.h
          /usr/local/include/google/protobuf/extension_set.h
          /usr/local/include/google/protobuf/field_mask.pb.h
          /usr/local/include/google/protobuf/generated_enum_reflection.h
          /usr/local/include/google/protobuf/generated_enum_util.h
          /usr/local/include/google/protobuf/generated_message_reflection.h
          /usr/local/include/google/protobuf/generated_message_util.h
          /usr/local/include/google/protobuf/io/coded_stream.h
          /usr/local/include/google/protobuf/io/gzip_stream.h
          /usr/local/include/google/protobuf/io/printer.h
          /usr/local/include/google/protobuf/io/strtod.h
          /usr/local/include/google/protobuf/io/tokenizer.h
          /usr/local/include/google/protobuf/io/zero_copy_stream.h
          /usr/local/include/google/protobuf/io/zero_copy_stream_impl.h
          /usr/local/include/google/protobuf/io/zero_copy_stream_impl_lite.h
          /usr/local/include/google/protobuf/map.h
          /usr/local/include/google/protobuf/map_entry.h
          /usr/local/include/google/protobuf/map_entry_lite.h
          /usr/local/include/google/protobuf/map_field.h
          /usr/local/include/google/protobuf/map_field_inl.h
          /usr/local/include/google/protobuf/map_field_lite.h
          /usr/local/include/google/protobuf/map_type_handler.h
          /usr/local/include/google/protobuf/message.h
          /usr/local/include/google/protobuf/message_lite.h
          /usr/local/include/google/protobuf/metadata.h
          /usr/local/include/google/protobuf/reflection.h
          /usr/local/include/google/protobuf/reflection_ops.h
          /usr/local/include/google/protobuf/repeated_field.h
          /usr/local/include/google/protobuf/repeated_field_reflection.h
          /usr/local/include/google/protobuf/service.h
          /usr/local/include/google/protobuf/source_context.pb.h
          /usr/local/include/google/protobuf/struct.pb.h
          /usr/local/include/google/protobuf/stubs/atomic_sequence_num.h
          /usr/local/include/google/protobuf/stubs/atomicops.h
          /usr/local/include/google/protobuf/stubs/atomicops_internals_arm64_gcc.h
          /usr/local/include/google/protobuf/stubs/atomicops_internals_arm_gcc.h
          /usr/local/include/google/protobuf/stubs/atomicops_internals_arm_qnx.h
          /usr/local/include/google/protobuf/stubs/atomicops_internals_atomicword_compat.h
          /usr/local/include/google/protobuf/stubs/atomicops_internals_generic_gcc.h
          /usr/local/include/google/protobuf/stubs/atomicops_internals_macosx.h
          /usr/local/include/google/protobuf/stubs/atomicops_internals_mips_gcc.h
          /usr/local/include/google/protobuf/stubs/atomicops_internals_pnacl.h
          /usr/local/include/google/protobuf/stubs/atomicops_internals_power.h
          /usr/local/include/google/protobuf/stubs/atomicops_internals_solaris.h
          /usr/local/include/google/protobuf/stubs/atomicops_internals_tsan.h
          /usr/local/include/google/protobuf/stubs/atomicops_internals_x86_gcc.h
          /usr/local/include/google/protobuf/stubs/atomicops_internals_x86_msvc.h
          /usr/local/include/google/protobuf/stubs/bytestream.h
          /usr/local/include/google/protobuf/stubs/callback.h
          /usr/local/include/google/protobuf/stubs/casts.h
          /usr/local/include/google/protobuf/stubs/common.h
          /usr/local/include/google/protobuf/stubs/fastmem.h
          /usr/local/include/google/protobuf/stubs/hash.h
          /usr/local/include/google/protobuf/stubs/logging.h
          /usr/local/include/google/protobuf/stubs/macros.h
          /usr/local/include/google/protobuf/stubs/mutex.h
          /usr/local/include/google/protobuf/stubs/once.h
          /usr/local/include/google/protobuf/stubs/platform_macros.h
          /usr/local/include/google/protobuf/stubs/port.h
          /usr/local/include/google/protobuf/stubs/scoped_ptr.h
          /usr/local/include/google/protobuf/stubs/shared_ptr.h
          /usr/local/include/google/protobuf/stubs/singleton.h
          /usr/local/include/google/protobuf/stubs/status.h
          /usr/local/include/google/protobuf/stubs/stl_util.h
          /usr/local/include/google/protobuf/stubs/stringpiece.h
          /usr/local/include/google/protobuf/stubs/template_util.h
          /usr/local/include/google/protobuf/stubs/type_traits.h
          /usr/local/include/google/protobuf/text_format.h
          /usr/local/include/google/protobuf/timestamp.pb.h
          /usr/local/include/google/protobuf/type.pb.h
          /usr/local/include/google/protobuf/unknown_field_set.h
          /usr/local/include/google/protobuf/util/field_comparator.h
          /usr/local/include/google/protobuf/util/field_mask_util.h
          /usr/local/include/google/protobuf/util/json_util.h
          /usr/local/include/google/protobuf/util/message_differencer.h
          /usr/local/include/google/protobuf/util/time_util.h
          /usr/local/include/google/protobuf/util/type_resolver.h
          /usr/local/include/google/protobuf/util/type_resolver_util.h
          /usr/local/include/google/protobuf/wire_format.h
          /usr/local/include/google/protobuf/wire_format_lite.h
          /usr/local/include/google/protobuf/wire_format_lite_inl.h
          /usr/local/include/google/protobuf/wrappers.pb.h
        
        Warning: Unbrewed .la files were found in /usr/local/lib.
        If you didn't put them there on purpose they could cause problems when
        building Homebrew formulae, and may need to be deleted.
        
        Unexpected .la files:
          /usr/local/lib/libprotobuf-lite.la
          /usr/local/lib/libprotobuf.la
          /usr/local/lib/libprotoc.la
        
        Warning: Unbrewed .pc files were found in /usr/local/lib/pkgconfig.
        If you didn't put them there on purpose they could cause problems when
        building Homebrew formulae, and may need to be deleted.
        
        Unexpected .pc files:
          /usr/local/lib/pkgconfig/protobuf-lite.pc
          /usr/local/lib/pkgconfig/protobuf.pc
        
        Warning: Unbrewed static libraries were found in /usr/local/lib.
        If you didn't put them there on purpose they could cause problems when
        building Homebrew formulae, and may need to be deleted.
        
        Unexpected static libraries:
          /usr/local/lib/libprotobuf-lite.a
          /usr/local/lib/libprotobuf.a
          /usr/local/lib/libprotoc.a
        
        Warning: You have unlinked kegs in your Cellar.
        Leaving kegs unlinked can lead to build-trouble and cause brews that depend on
        those kegs to fail to run properly once built. Run `brew link` on these:
          gettext
          protobuf
        
        Warning: Homebrew's sbin was not found in your PATH but you have installed
        formulae that put executables in /usr/local/sbin.
        Consider setting the PATH for example like so:
          echo 'export PATH="/usr/local/sbin:$PATH"' >> ~/.zshrc
        
        Warning: Some installed formulae are missing dependencies.
        You should `brew install` the missing dependencies:
          brew install openjdk
        
        Run `brew missing` for more details.
`

**Comment by neutric** [comment]
What should be
```
HOMEBREW_VERSION: 3.3.4-72-g27a7629
ORIGIN: https://github.com/Homebrew/brew
HEAD: 27a76295a4dede994509b0c5b9120e7e38b8ffc4
Last commit: 24 hours ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core

```
is on your side
```
HOMEBREW_VERSION: 2.5.2
ORIGIN: https://mirrors.aliyun.com/homebrew/brew.git
HEAD: cd4a3c0588abb82d916bc8558f4c157e951d449d
Last commit: 1 year, 2 months ago 
Core tap ORIGIN: https://mirrors.aliyun.com/homebrew/homebrew-core.git
```

`ORIGIN: https://mirrors.aliyun.com/homebrew/brew.git` must have grown tired of mirroring.

**Reply by gromgit** [reply]
Actually, onyx is a cask, and cask config isn't listed in the `brew config` output. I asked because I suspected (and it was confirmed) that `brew update` hadn't been run in a while. (The obsolete `brew cask install` command was another clue.)

@whoooami, did you run `brew update` before `brew config`? If so, the aliyun mirrors are stuck in the year 2020, and you should stop using them. The `brew doctor` output tells you what to do now:
```
[...]
git -C "/usr/local/Homebrew" remote set-url origin https://github.com/Homebrew/brew
[...]
git -C "/usr/local/Homebrew/Library/Taps/homebrew/homebrew-core" remote set-url origin https://github.com/Homebrew/homebrew-core
[...]
brew untap caskroom/cask
```
Do all that, then the following:
```
brew update-reset
brew install --cask onyx
```
Note the new syntax for cask installs.

**Reply by SMillerDev** [reply]
Looking at that mirror they are still going and @whoooami just hasn't updated in over a year.

**Reply by carlocab** [reply]
Yep
```
❯ git -C "$(brew --repo)" show "$(git ls-remote https://mirrors.aliyun.com/homebrew/brew.git HEAD | cut -f1)"
commit 27a76295a4dede994509b0c5b9120e7e38b8ffc4 (origin/master, origin/HEAD)
Merge: 7241f8db2 ce2d946a7
Author: Alexander Bayandin <a.bayandin@gmail.com>
Date:   Fri Nov 19 12:24:33 2021 +0000

    Merge pull request #12279 from bayandin/synced-versions-formulae

    formula_auditor: add audit for formulae with synced versions
```

**Reply by whoooami** [reply]
I change to official address. but any request was time out like this
`
     ==> Fetching /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core...

> remote: Enumerating objects: 305029, done.
> remote: Counting objects: 100% (305029/305029), done.
> remote: Compressing objects: 100% (84691/84691), done.
> remote: Total 300090 (delta 226780), reused 288527 (delta 215349), pack-reused 0
> Receiving objects: 100% (300090/300090), 122.67 MiB | 2.94 MiB/s, done.
> Resolving deltas: 100% (226780/226780), completed with 4889 local objects.
> From https://github.com/Homebrew/homebrew-core
>    136a43943a8..0c0e75a99d7  master     -> origin/master
> fatal: unable to access 'https://github.com/Homebrew/homebrew-core/': LibreSSL SSL_connect: SSL_ERROR_SYSCALL in connection to github.com:443
`
my problem. Maybe I will fix the GFW problem in the future.
Thx all.

**Comment by whoooami** [comment]
> What should be
> 
> ```
> HOMEBREW_VERSION: 3.3.4-72-g27a7629
> ORIGIN: https://github.com/Homebrew/brew
> HEAD: 27a76295a4dede994509b0c5b9120e7e38b8ffc4
> Last commit: 24 hours ago
> Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
> ```
> 
> is on your side
> 
> ```
> HOMEBREW_VERSION: 2.5.2
> ORIGIN: https://mirrors.aliyun.com/homebrew/brew.git
> HEAD: cd4a3c0588abb82d916bc8558f4c157e951d449d
> Last commit: 1 year, 2 months ago 
> Core tap ORIGIN: https://mirrors.aliyun.com/homebrew/homebrew-core.git
> ```
> 
> `ORIGIN: https://mirrors.aliyun.com/homebrew/brew.git` must have grown tired of mirroring.

I change to official address. but any request was time out like this
`
     ==> Fetching /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core...

> remote: Enumerating objects: 305029, done.
> remote: Counting objects: 100% (305029/305029), done.
> remote: Compressing objects: 100% (84691/84691), done.
> remote: Total 300090 (delta 226780), reused 288527 (delta 215349), pack-reused 0
> Receiving objects: 100% (300090/300090), 122.67 MiB | 2.94 MiB/s, done.
> Resolving deltas: 100% (226780/226780), completed with 4889 local objects.
> From https://github.com/Homebrew/homebrew-core
>    136a43943a8..0c0e75a99d7  master     -> origin/master
> fatal: unable to access 'https://github.com/Homebrew/homebrew-core/': LibreSSL SSL_connect: SSL_ERROR_SYSCALL in connection to github.com:443
`
my problem. Maybe I will fix the GFW problem in the future.

**Reply by SMillerDev** [reply]
You can keep using the other address. You just need to run `brew update` or `brew update-reset` because your homebrew version is very far behind.

**Reply by whoooami** [reply]
Yeah, thank you! I will try again.
